src/HypothesisClient/Client/Users.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace eLife\HypothesisClient\Client;
+
+use eLife\HypothesisClient\ApiClient\UsersClient;
+use eLife\HypothesisClient\Exception\BadResponse;
+use eLife\HypothesisClient\Model\User;
+use eLife\HypothesisClient\Result\ResultInterface;
+use GuzzleHttp\Promise\PromiseInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use function GuzzleHttp\Promise\exception_for;
+
+final class Users implements ClientInterface
+{
+    private $normalizer;
+    private $usersClient;
+
+    public function __construct(UsersClient $usersClient, DenormalizerInterface $normalizer)
+    {
+        $this->usersClient = $usersClient;
+        $this->normalizer = $normalizer;
+    }
+
+    public function get(string $id) : PromiseInterface
+    {
+        return $this->usersClient
+            ->getUser(
+                [],
+                $id
+            )
+            ->then(function (ResultInterface $result) {
+                return $this->normalizer->denormalize($result->toArray(), User::class);
+            });
+    }
+
+    /**
+     * Store the user by create first then, if user already detected, modify.
+     *
+     * @param User $user
+     *
+     * @return PromiseInterface
+     */
+    public function store(User $user) : PromiseInterface
+    {
+        return $this->create($user)
+            ->otherwise(function ($reason) use ($user) {
+                $exception = exception_for($reason);
+                if ($exception instanceof BadResponse && preg_match('/user with username [^\s]+ already exists/', (string) $exception->getResponse()->getBody())) {
```

Comment:
That's a shame, there's no better way to update/pick out that error?

Reply:
I'm considering just reacting to the `BadResponse`. Potentially, it might let one through that we could have also expected to fail for the patch request but the check would be less fragile. The most likely cause for the `Bad Request` will be that the username already exists.